lib/resque/worker.rb

```diff
@@ -861,7 +862,11 @@ def perform_with_fork(job, &block)
 
       begin
         @child = fork do
-          unregister_signal_handlers if term_child
+          if term_child
+            unregister_signal_handlers
+          else
+            @signal_handler.reopen
```

Comment:
It is because

1. fork does not copy thread
2. IO.pipe is shared with the parent process

which results in that receiving a signal in child process makes an effect to its parent process.
`reopen` newly creates signal handler thread and pipe for the child process.